src/utils/Configuration.ts

```diff
@@ -356,17 +392,38 @@ export default class Configuration {
     }
   }
 
+  private static getUserProvidedCredential(key: CloudCredentialsKey): string {
+    // Get the credentials
+    const credentials: CloudCredentials = Configuration.getAppEnv().getServiceCreds('emobility-credentials') as CloudCredentials;
+    if (!Configuration.isNullOrUndefined(credentials) && !Configuration.isUndefined(credentials[key])) {
+      return credentials[key];
+    }
+    console.error(`Credentials or credential key ${key} not found or bound in CF 'emobility-credentials' User Provided Credentials Service`);
+  }
+
   // Read the config file
   private static getConfig(): ConfigurationData {
-    if (!this.config) {
-      this.config = JSON.parse(fs.readFileSync(`${global.appRoot}/assets/config.json`, 'utf8')) as ConfigurationData;
+    if (!Configuration.config) {
+      Configuration.config = JSON.parse(fs.readFileSync(`${global.appRoot}/assets/config.json`, 'utf8')) as ConfigurationData;
+    }
+    return Configuration.config;
+  }
+
+  private static getAppEnv(): AppEnv {
+    if (!Configuration.appEnv) {
+      Configuration.appEnv = getAppEnv();
```

Comment:
Weird, same method name here 'getAppEnv'.
```
private static getAppEnv(): AppEnv {
    if (!Configuration.appEnv) {
      Configuration.appEnv = getAppEnv();
```
Have you tested the call?

Reply:
Yes, there's no name collision and the deployment test on the dev landscape are working fine. Only the push notification have not been tested. So I think that PR should only be merged:
+ when you plan to deploy only on the QA for more tests coverage;
+ when the UPS setup is done on all landscape;
+ when the SCP services dependency is updated on all manifest files for dev and QA landscape. And then prod;